server/jetstream_api.go

```diff
@@ -581,6 +581,7 @@ const JSApiLeaderStepDownResponseType = "io.nats.jetstream.api.v1.meta_leader_st
 type JSApiMetaServerRemoveRequest struct {
 	// Server name of the peer to be removed.
 	Server string `json:"peer"`
+	ByID   bool   `json:"by_id,omitempty"`
```

Comment:
Just `Peer`

Reply:
Why not take whats given in `peer` and try to use it as a name, if it fails try to use it as a ID and if fails errors

Reply:
I thought about that, and we would not need to update the request field, which is nice, but then I got concerned that what if a server name is equal to the peer ID (the hash) of another server. Say server name "A" has id "B" (I know it is 8 chars, but you get the idea) and server "B"'s id is "A". It would have to be VERY clear that the name provided to the peer-remove command is by "server name" first, and only if not found would we remove looking by ID. Because if the user uses the "raft peer-remove B" and assumes that B is the peer ID, then instead of remove server "A" (whose ID is "B"), the server "B" (id "A") is being removed. Sounds crazy because very unlikely that peer ID and server names ever cross-match, but since the command is "peer-remove" (not server-remove), for a new user it could be possible to make the mistake, especially if we do provide the "peer ID" as a new column.

Don't get me wrong, I like your approach and that simplifies a lot (and does not need a CLI change), but I wanted to raise the far-fetched concern above. Especially if we start returning peer ID to the CLI that would display it as a column. One could say that if we do what you propose maybe we don't need to return explicitly the peer ID in PeerInfo objects, since the Name has the peerID info for the cases where it is not known. What do you think?

Reply:
Yeah that's a valid concern - and why I like the restriction to only remove offline nodes.  But yes, of course you could have the situation also with offline peers.

I don't want to parse that sentence to extract the peer id (again thinking about the case of presenting a list of nodes)

Reply:
So you would still be ok to have a command that explicitly ask to remove by ID? (so update of the request struct, etc..)

Reply:
Perhaps, but are we sure a clash in name/id wont be a problem later? Is uniqueness not something in these things not something we should care about?

Reply:
But I am saying that if the CLI has a new EXPLICIT command to remove by id (instead of name), then the struct will have "Peer string" and the server will only look for raft peer ID (not server names). So then it is no longer a problem.

Reply:
Meaning that you have 2 ways to remove: by name or by id, then since it is explicit there is no risk of clash. You can still have typos/messing up, but the same way right now if you remove by name and pick the wrong one :-)